src/main/java/emt/tile/TileEntityEtherealMacerator.java

```diff
@@ -226,18 +225,17 @@ else if (slots[i].stackSize < 63)
         }
     }
 
-    public boolean isOP() {
-        final int cobblestoneId = OreDictionary.getOreID("blockCobble");
+    public boolean isAllowed() {
 
-        if (OreDictionary.getOreID(this.slots[0]) == cobblestoneId || this.slots[0].getItem() == Item.getItemFromBlock(Blocks.sand) || this.slots[0].getItem() == Items.potato || this.slots[0].getItem() == Items.porkchop || this.slots[0].getItem() == Items.beef || this.slots[0].getItem() == Items.chicken || this.slots[0].getItem() == Items.fish
-                || this.slots[0].getItem() == IC2Items.getItem("coalDust").getItem()
-                || (OreDictionary.getOreName(OreDictionary.getOreID(this.slots[0])).toLowerCase().contains("dust") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("armour") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("armor") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("helm")
-                || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("legging") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("boot") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("chestp") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("pick")
-                || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("shovel") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("axe") || this.slots[0].getItem().getUnlocalizedName(this.slots[0]).toLowerCase().contains("pick"))) {
-            return true;
-        } else {
-            return false;
+        int[] OreDictIDs = OreDictionary.getOreIDs(this.slots[0]);
+        for (int ID : OreDictIDs) {
+            String tName = OreDictionary.getOreName(ID);
+            for (String allowed : allowedOreDictList) {
```

Comment:
String comparison is more expansive than integer comparison. Please prepare an `int[] allowedOreDictIds` array somewhere and initialize it the first time this array is needed, instead of doing all these string lookups. Preferably sort that array so we can use `Arrays.binarySearch` instead of going over the whole array.

Reply:
i can't change it into int because the allowedOreDictList are only prefix of the OreDict